api/server/router/system/system_routes.go

```diff
@@ -90,49 +101,80 @@ func (s *systemRouter) getVersion(ctx context.Context, w http.ResponseWriter, r
 }
 
 func (s *systemRouter) getDiskUsage(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
-	eg, ctx := errgroup.WithContext(ctx)
-
-	var du *types.DiskUsage
-	eg.Go(func() error {
-		var err error
-		du, err = s.backend.SystemDiskUsage(ctx)
+	if err := httputils.ParseForm(r); err != nil {
 		return err
-	})
+	}
+
+	var getContainers, getImages, getVolumes, getBuildCache bool
+	typesStrs, ok := r.Form["types"]
+	if !ok {
+		getContainers, getImages, getVolumes, getBuildCache = true, true, true, true
+	} else {
+		for _, s := range typesStrs {
+			for _, typ := range strings.Split(s, ",") {
+				switch typ {
+				case "container", "containers":
```

Comment:
I'd omit the multiple variants; it's better to be strict here. It's an API, so it's ok to have a fixed list of options (if we want aliases for "convenience", those can be added in the CLI if we think the UX can be improved by adding them.

Perhaps would be nice to have `const`s for these values, so that a client can use those 🤔